source/lambda/use-case-management/model/use-case.ts

```diff
@@ -296,25 +297,47 @@ export class ChatUseCaseDeploymentAdapter extends UseCase {
         );
         cfnParameters.set(CfnParameterKeys.UseCaseConfigTableName, process.env[USE_CASE_CONFIG_TABLE_NAME_ENV_VAR]!);
 
-        cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, process.env[USER_POOL_ID_ENV_VAR]!);
-        if (
-            process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR] &&
-            process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR].toLowerCase() === 'true' &&
-            process.env[CLIENT_ID_ENV_VAR]
-        ) {
-            cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolClient, process.env[CLIENT_ID_ENV_VAR]);
+        if (eventBody.AuthenticationParams) {
+            switch (eventBody.AuthenticationParams.AuthenticationProvider) {
+                case AUTHENTICATION_PROVIDERS.COGNITO:
```

Comment:
for the sake of full safety, maybe consider a default case with an error logging that an unsupported type was provided